### ISSUE TITLE
Remove browserslist, leave processing and transpilation to applications

### ISSUE THREAD

**Issue by VsevolodGolovanov** [primefaces/primereact#1312]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

Seems like currently styles have vendor prefixed rules specified by hand. This leads to inconsistencies. E.g. this flex-related stuff has IE10 prefixed version:

```
.p-fluid .p-inputgroup .p-inputtext {
    -webkit-box-flex: 1;
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
}
```

But this doesn't:

```
.p-dialog {
    display: flex;
    flex-direction: column;
    padding: 0;
    pointer-events: auto;
    max-height: 90%;
}
```

As a result dialogs don't become visible in IE10.

But Create React App uses PostCSS with Autoprefixer plugin and others.
If I just specify in my app css:

```
.p-dialog-visible {
    display: flex;
}
```

when built this turns into

```
.p-dialog-visible{display:-ms-flexbox;display:flex}
```

and now the dialog is actually shown in IE10.

But gotta be careful to keep CSS properly scoped to components. E.g. unlike CRA PrimeReact probably shouldn't use normalize and other things that affect global rules.

**Comment by VsevolodGolovanov** [comment]
I was under the impression that CRA is set up in such a way, that my CSS (that gets built into `main.<hash>.chunk.css`) gets processed, but 3rd party (PrimeReacts's, `2.<hash>.chunk.css`) doesn't.

[Turns out](https://stackoverflow.com/q/61144331/1341535) that is not the case at all. 3rd party CSS gets processed as well, but PrimeReact's stricter browserslist overrides my more lenient rules.

Should a component library do this? Of course, it's fine to officially support certain browser versions. But maybe that's better specified in a readme than a browserslist?

Microsoft's Fabric/Fluent UI, Semantic UI, Material UI: all don't have browserslist in their package.json (maybe their configs are organized differently though, I don't know). Ant Design's component library does.

**Comment by VsevolodGolovanov** [comment]
Renamed the issue accordingly.

I assume JS transpilation works the same way - that it affects 3rd party code too.

**Comment by cagataycivici** [comment]
We're modernizing the css of PrimeReact these days targeting upcoming 5.0, where CSS or every component will be reviewed and modernized. We'll review this next week when the work begins, thank you for the feedback.

**Comment by mertsincan** [comment]
Fixed for 5.0.

Best Regards,

**Comment by VsevolodGolovanov** [comment]
@mertsincan , I'm not sure, what was fixed here? PrimeReact's package.json still contains the same, very limiting, browserslist.
